packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -94,11 +94,18 @@ export class HardhatNetworkProvider extends EventEmitter
     }
 
     try {
+      let result;
       if (this._loggingEnabled && !PRIVATE_RPC_METHODS.has(args.method)) {
-        return await this._sendWithLogging(args.method, args.params);
+        result = await this._sendWithLogging(args.method, args.params);
+      } else {
+        result = await this._send(args.method, args.params);
+      }
+
+      if (args.method === "hardhat_reset") {
+        this.emit("hardhat_reset");
```

Comment:
Shouldn't we emit it in the node and just fwd it in the provider?

Reply:
As far as I understand, a new node is created on a reset, so that wouldn't be possible, would it?